modules/rosetta/src/main/scala/org/tessellation/rosetta/http/routes/ConstructionRoutes.scala

```diff
@@ -44,6 +44,23 @@ final class ConstructionRoutes[F[_]: Async](
 
         Ok(ConstructionPreprocess.Response(accountIdentifiers)).handleUnknownError
       }
+
+    case req @ POST -> Root / "parse" =>
+      req
+        .decodeRosettaWithNetworkValidation[ConstructionParse.Request](appEnvironment, _.networkIdentifier) { parseReq =>
+          constructionService
+            .parseTransaction(parseReq.transaction, parseReq.signed)
+            .bimap(
+              _.toRosettaError,
+              serviceResponse =>
+                ConstructionParse.Response(
+                  serviceResponse.operations.toList,
+                  serviceResponse.accountIdentifierSigners
+                )
```

Comment:
in `ConstructionParse.Response` you can write a fn:
```scala
def fromParseResult(result: ParseResult): Response = ...
```
and you can use it here like:
```scala
.bimap(
  _.toRosettaError,
  ConstructionParse.Response.fromParseResult
)
```

Reply:
Changed.